system_baseline/views/v1.py

```diff
@@ -54,10 +51,17 @@ def get_baselines_by_ids(baseline_ids, limit, offset, order_by, order_how):
     query = SystemBaseline.query.filter(
         SystemBaseline.account == account_number, SystemBaseline.id.in_(baseline_ids)
     )
-    try:
-        ensure_correct_system_count(baseline_ids, query.all())
-    except ItemNotReturned as e:
-        raise HTTPError(HTTPStatus.NOT_FOUND, message=e.message)
+    if len(query.all()) < len(baseline_ids):
```

Comment:
you may be able to swap `len(query.all())` for `query.count()`

Reply:
actually, I would leave it as query.all(), but do something like this:

```python
full_results = query.all()
if len(full_results) < len(baseline_ids):
  fetched_ids = {result.id for result in full_results}
  missing_ids = set(baseline_ids) - fetched_ids
  # raise not_found error here with join(missing_ids)
```

There are other ways to do it too but this avoids making multiple queries and lets python do more work via set magic